app/forms/delivery_partner_form.rb

```diff
@@ -86,10 +86,9 @@ def lead_providers_validation
   def cohorts_validation
     # Ensure all selected lead providers have at least one selected cohort
     # This is indicated by the presence of a provider relationship for that lead provider
-    lead_provider_ids&.filter(&:present?)&.each do |lead_provider_id|
+    lead_provider_ids&.filter(&:present?)&.any? do |lead_provider_id|
       unless chosen_provider_relationships.pluck(:lead_provider_id).include?(lead_provider_id)
         errors.add(:provider_relationship_hashes, :blank, message: "Choose at least one cohort for every lead provider")
-        break
```

Comment:
What happens if we have have two `lead_provider_ids `?

Do we get duplicate error?

Reply:
Also, in principle I dislike side effects in functions like `map`, `filter`, and yes, `any` - `each` is the 'principled' functional array method that is designed to allow for those.

Maybe a slightly more semantically correct way of going about it would be
```
cohorts_selected =  lead_provider_ids&.filter(&:present?)&.all? do | lead_provider_id | 
  chosen_provider_relationships.pluck(:lead_provider_id).include?(lead_provider_id)}
end

errors.add(:provider_relationship_hashes, :blank, message: "Choose at least one cohort for every lead provider") unless cohorts_selected
```

Thoughts?

Reply:
> Do we get duplicate error?
I would assume that `any?` short-circuits and so prevents that. I prefer your suggestion, however

Reply:
The former stops as soon as it hits the condition, the latter runs them all. I prefer the former as closer to the original. 
If you wanted to have a "true" at the end of the block to make it less of a side-effect.

Reply:
I can see the argument for `any` being more of a functional list/comprehension type thing so having procedural stuff in it being a bit odd, but given it currently breaks the coverage I'd probably take it as is and then polish later if we feel like it.

The use of `all` will likely be less performant than `any` as it has to run for the whole set rather than exit early.

Reply:
how about using `first` and then checking for whether that returned anything....

Reply:
I think the use of `errors.add` to generate a `true` for the `unless` is a bit hard to follow, I'll try and make something I like better....

Reply:
Performance would be my last concern here, we are talking about order of magnitude of '6 lead providers' 😂

Reply:
Excuse the deleted comments, I had to play with the code to really grok this.

I agree with @kicferk1 on the suggested change, sorry @blackrat !

Reply:
I dislike the variable usage. I'm more inclined to pull that into a different method and refer to it that way. Thought?

Reply:
I don't mind putting the code checking if there is a cohort selected for each LP into its own method.

Reply:
Actually, I had some time to put more thought into this. It was a quick first thing that worked with minimal code, but in thinking about it, it just comes down to an array comparison. If there are any elements in the first array that aren't in the second element, then we have an error. i.e. if (A-B) is not empty add the error in, otherwise it's fine.

Reply:
Set comparison, perhaps, but it's possible to have multiple copies of the lead provider in chosen_provider_relationships